src/d3a/d3a_core/simulation.py

```diff
@@ -539,6 +543,40 @@ def __setstate__(self, state):
         self.__dict__.update(state)
         self._load_setup_module()
 
+    def loop_area_object(self, target_area):
+        duplicate_values = ""
+        try:
+            if target_area.parent is None:
+                duplicate_values = self.get_duplicate_data(target_area.children)
+                if len(duplicate_values) > 0:
+                    self.rename_area_object(target_area.children, duplicate_values)
+            for child in target_area.children:
+                if len(child.children) > 0:
+                    duplicate_values = self.get_duplicate_data(child.children)
+                if len(duplicate_values) > 0:
+                    self.rename_area_object(child.children, duplicate_values)
+                    self.loop_area_object(child)
+        except D3AException as ex:
+            print_exc()
+            raise D3AException("Cannot loop over the area object for renaming", ex)
```

Comment:
```suggestion
    def loop_area_object(self, target_area):
        if len(target_area.children) > 0:
            duplicate_values = self.get_duplicate_data(target_area.children)
            if len(duplicate_values) > 0:
                    self.rename_area_object(target_area.children, duplicate_values)
            for child in target_area.children:
                    self.loop_area_object(child)
                    
                
                    

```

Reply:
changed and added parameter as list to rename